engine/src/graphics/pipeline.cpp

```diff
@@ -0,0 +1,182 @@
+// Copyright (c) 2023-present Genesis Engine contributors (see LICENSE.txt)
+
+#include "graphics/pipeline.hpp"
+#include "graphics/vkHelpers.hpp"
+
+#include <array>
+#include <filesystem>
+#include <fstream>
+#include <stdexcept>
+#include <string>
+#include <vector>
+
+namespace gen
+{
+	GraphicsPipeline::GraphicsPipeline(GraphicsDevice & device, const PipelineConfigInfo & configInfo)
+		: m_device{device}, m_configInfo{configInfo}
+	{
+		createGraphicsPipeline();
+		m_logger.info("Graphics pipeline constructed");
+	}
+
+	GraphicsPipeline::~GraphicsPipeline()
+	{
+		m_logger.info("Graphics pipeline destructed");
+	}
+
+	PipelineConfigInfo GraphicsPipeline::defaultPipelineConfigInfo(const mim::vec2i& extent)
+	{
+		PipelineConfigInfo configInfo{};
+
+		configInfo.viewport.x		 = 0.0F;
+		configInfo.viewport.y		 = 0.0F;
+		configInfo.viewport.width	 = static_cast<float>(extent.x);
+		configInfo.viewport.height	 = static_cast<float>(extent.y);
```

Comment:
Recommend [flipping the viewport](https://www.saschawillems.de/blog/2019/03/29/flipping-the-vulkan-viewport/).